<a href="https://colab.research.google.com/github/AlexiasXro/tendencias-empleo-ia/blob/main/Laboratorio1_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Analisis de Datos: "Tendencias del Empleo en Inteligencia Artificial"

#### Analista de Datos: A. Romina Cáceres
#### Dataset: [pagina kaggle](https://www.kaggle.com/datasets/abhishekjaiswal4896/ai-job-market-trends)


## 1 - ETL Extración, Tranformación y Carga

### 1. Extración

In [3]:

from google.colab import drive
import pandas as pd
import chardet #libreria para los encoding

#Mostramos Google Drive para acceder al archivo
drive.mount('/content/drive', force_remount=True)

ruta_archivo = '/content/drive/MyDrive/Analisis de datos 2025/Tendencias del mercado laboral de IA/ai_job_market.csv'

with open(ruta_archivo, 'rb') as f:
  #podemos limitar los kbites f.read(5000)
  result = chardet.detect(f.read())


try:
  df = pd.read_csv(ruta_archivo, encoding=result['encoding'], sep=',', decimal='.')

  print('Extración de datos exitosa')
except FileNotFoundError:
  print('El archivo no se encuentra en la ruta especificada')




Mounted at /content/drive
Extración de datos exitosa


### 2.   Transformación



In [67]:
#   estructura general
df.info()
print("- OBSERVACIÓN: 11 son de tipo object, lo que indica que contienen texto o cadenas de caracteres.Solo 1 columna (job_id) es de tipo int64, es decir, numérica entera.")
df.head()
df.describe(include='all')




<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   job_id                 2000 non-null   int64         
 1   Empresa                2000 non-null   object        
 2   Industria              2000 non-null   object        
 3   Puesto                 2000 non-null   object        
 4   Habilidades            2000 non-null   object        
 5   Experiencia            2000 non-null   object        
 6   Tipo Empleo            2000 non-null   object        
 7   Ubicación              2000 non-null   object        
 8   Rango Salarial         2000 non-null   object        
 9   Fecha Publicación      2000 non-null   datetime64[ns]
 10  Herramientas           2000 non-null   object        
 11  Salario_Min            2000 non-null   int64         
 12  Salario_Max            2000 non-null   int64         
 13  Sal

job_id      Empresa   Industria        Puesto  \
count   2000.000000         2000        2000          2000   
unique          NaN         1909           7             8   
top             NaN  Johnson LLC  Automotive  Data Analyst   
freq            NaN            4         300           271   
mean    1000.500000          NaN         NaN           NaN   
min        1.000000          NaN         NaN           NaN   
25%      500.750000          NaN         NaN           NaN   
50%     1000.500000          NaN         NaN           NaN   
75%     1500.250000          NaN         NaN           NaN   
max     2000.000000          NaN         NaN           NaN   
std      577.494589          NaN         NaN           NaN   

                        Habilidades Experiencia Tipo Empleo  \
count                          2000        2000        2000   
unique                         1985           3           4   
top     Azure, Excel, Power BI, SQL       Entry  Internship   
freq                              2         702         574   
mean                            NaN         NaN         NaN   
min                             NaN         NaN         NaN   
25%                             NaN         NaN         NaN   
50%                             NaN         NaN         NaN   
75%                             NaN         NaN         NaN   
max                             NaN         NaN         NaN   
std                             NaN         NaN         NaN   

                     Ubicación Rango Salarial    Fecha Publicación  \
count                     2000           2000                 2000   
unique                    2000           2000                  NaN   
top     North Melissashire, LA  126658-206526                  NaN   
freq                         1              1                  NaN   
mean                       NaN            NaN  2024-09-21 15:00:00   
min                        NaN            NaN  2023-09-21 00:00:00   
25%                        NaN            NaN  2024-03-20 18:00:00   
50%                        NaN            NaN  2024-09-26 00:00:00   
75%                        NaN            NaN  2025-03-16 00:00:00   
max                        NaN            NaN  2025-09-19 00:00:00   
std                        NaN            NaN                  NaN   

       Herramientas    Salario_Min    Salario_Max  Salario_Promedio  \
count          2000    2000.000000    2000.000000       2000.000000   
unique          352            NaN            NaN               NaN   
top            KDB+            NaN            NaN               NaN   
freq             96            NaN            NaN               NaN   
mean            NaN   95374.602000  150705.401000     123040.001500   
min             NaN   40044.000000   54162.000000      47578.500000   
25%             NaN   68234.500000  120751.250000      94466.250000   
50%             NaN   95440.000000  151337.500000     123202.750000   
75%             NaN  123580.500000  182200.000000     151413.000000   
max             NaN  149975.000000  247578.000000     197776.500000   
std             NaN   31882.260505   41716.198452      34776.200983   

        Año_Publicación  Mes_Publicación Tamaño_Estimado  \
count       2000.000000      2000.000000            2000   
unique              NaN              NaN               2   
top                 NaN              NaN     Desconocido   
freq                NaN              NaN            1343   
mean        2024.226000         6.499000             NaN   
min         2023.000000         1.000000             NaN   
25%         2024.000000         3.000000             NaN   
50%         2024.000000         7.000000             NaN   
75%         2025.000000        10.000000             NaN   
max         2025.000000        12.000000             NaN   
std            0.667942         3.499875             NaN   

        Cantidad_Herramientas  
count             2000.000000  
unique                    NaN  


 ** Explicación de estadísticas en df.describe()**

- **count:** cantidad de valores no nulos en la columna.
- **unique:** cantidad de valores distintos (*solo aplica a columnas categóricas).*
- **top:** valor más frecuente *(la moda)* en columnas categóricas.
- **freq:** frecuencia del valor más común *(cuántas veces aparece el 'top').*
- **mean:** promedio de los valores numéricos.
- **std:** desviación estándar, mide la dispersión respecto al promedio.
- ** min:** valor mínimo en la columna.
- ** 25%:** primer cuartil, el 25% de los datos están por debajo de este valor.
- **50%:** segundo cuartil o mediana, el valor central de la distribución.
- **75%:** tercer cuartil, el 75% de los datos están por debajo de este valor.
- **max:** valor máximo en la columna.



In [55]:
#Renombrando de columnas
df = df.rename(columns={
    'company_name': 'Empresa',
    'industry': 'Industria',
    'job_title': 'Puesto',
    'skills_required': 'Habilidades',
    'experience_level': 'Experiencia',
    'employment_type': 'Tipo Empleo',
    'location': 'Ubicación',
    'salary_range_usd': 'Rango Salarial',
    'posted_date': 'Fecha Publicación',
    'company_size': 'Tamaño Original',
    'tools_preferred': 'Herramientas',
    'Salario Min': 'Salario_Min',
    'Salario Max': 'Salario_Max',
    'Salario Promedio': 'Salario_Promedio',
    'Año': 'Año_Publicación',
    'Mes': 'Mes_Publicación',
    'Tamaño Empresa': 'Tamaño_Estimado',
    'Cantidad Herramientas': 'Cantidad_Herramientas'
})

# Convercion de fechas
df['Fecha Publicación'] = pd.to_datetime(df['Fecha Publicación'], errors='coerce')

# Convercion de numéricos
df['Salario_Min'] = pd.to_numeric(df['Salario_Min'], errors='coerce')
df['Salario_Max'] = pd.to_numeric(df['Salario_Max'], errors='coerce')
df['Salario_Promedio'] = pd.to_numeric(df['Salario_Promedio'], errors='coerce')
df['Cantidad_Herramientas'] = pd.to_numeric(df['Cantidad_Herramientas'], errors='coerce')

df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   job_id                 2000 non-null   int64         
 1   Empresa                2000 non-null   object        
 2   Industria              2000 non-null   object        
 3   Puesto                 2000 non-null   object        
 4   Habilidades            2000 non-null   object        
 5   Experiencia            2000 non-null   object        
 6   Tipo Empleo            2000 non-null   object        
 7   Ubicación              2000 non-null   object        
 8   Rango Salarial         2000 non-null   object        
 9   Fecha Publicación      2000 non-null   datetime64[ns]
 10  Herramientas           2000 non-null   object        
 11  Salario_Min            2000 non-null   int64         
 12  Salario_Max            2000 non-null   int64         
 13  Sal

In [58]:
# Eliminación de duplicados
df = df.drop_duplicates()

#  Ver cantidad de nulos por columna
df.isnull().sum()

#  Eliminación de filas con datos clave faltantes
df = df.dropna(subset=['Puesto', 'Rango Salarial'])

# Rellenado de valores faltantes en columnas categóricas
df['Tamaño_Estimado'] = df['Tamaño_Estimado'].fillna('Desconocido')
df['Experiencia'] = df['Experiencia'].fillna(df['Experiencia'].mode()[0])

#  Rellenado de valores faltantes en columnas numéricas
df['Salario_Promedio'] = df['Salario_Promedio'].fillna(df['Salario_Promedio'].mean())
df['Cantidad_Herramientas'] = df['Cantidad_Herramientas'].fillna(0)

# Estructura final
df.describe(include='all')



job_id      Empresa   Industria        Puesto  \
count   2000.000000         2000        2000          2000   
unique          NaN         1909           7             8   
top             NaN  Johnson LLC  Automotive  Data Analyst   
freq            NaN            4         300           271   
mean    1000.500000          NaN         NaN           NaN   
min        1.000000          NaN         NaN           NaN   
25%      500.750000          NaN         NaN           NaN   
50%     1000.500000          NaN         NaN           NaN   
75%     1500.250000          NaN         NaN           NaN   
max     2000.000000          NaN         NaN           NaN   
std      577.494589          NaN         NaN           NaN   

                        Habilidades Experiencia Tipo Empleo  \
count                          2000        2000        2000   
unique                         1985           3           4   
top     Azure, Excel, Power BI, SQL       Entry  Internship   
freq                              2         702         574   
mean                            NaN         NaN         NaN   
min                             NaN         NaN         NaN   
25%                             NaN         NaN         NaN   
50%                             NaN         NaN         NaN   
75%                             NaN         NaN         NaN   
max                             NaN         NaN         NaN   
std                             NaN         NaN         NaN   

                     Ubicación Rango Salarial    Fecha Publicación  \
count                     2000           2000                 2000   
unique                    2000           2000                  NaN   
top     North Melissashire, LA  126658-206526                  NaN   
freq                         1              1                  NaN   
mean                       NaN            NaN  2024-09-21 15:00:00   
min                        NaN            NaN  2023-09-21 00:00:00   
25%                        NaN            NaN  2024-03-20 18:00:00   
50%                        NaN            NaN  2024-09-26 00:00:00   
75%                        NaN            NaN  2025-03-16 00:00:00   
max                        NaN            NaN  2025-09-19 00:00:00   
std                        NaN            NaN                  NaN   

       Herramientas    Salario_Min    Salario_Max  Salario_Promedio  \
count          2000    2000.000000    2000.000000       2000.000000   
unique          352            NaN            NaN               NaN   
top            KDB+            NaN            NaN               NaN   
freq             96            NaN            NaN               NaN   
mean            NaN   95374.602000  150705.401000     123040.001500   
min             NaN   40044.000000   54162.000000      47578.500000   
25%             NaN   68234.500000  120751.250000      94466.250000   
50%             NaN   95440.000000  151337.500000     123202.750000   
75%             NaN  123580.500000  182200.000000     151413.000000   
max             NaN  149975.000000  247578.000000     197776.500000   
std             NaN   31882.260505   41716.198452      34776.200983   

        Año_Publicación  Mes_Publicación Tamaño_Estimado  \
count       2000.000000      2000.000000            2000   
unique              NaN              NaN               2   
top                 NaN              NaN     Desconocido   
freq                NaN              NaN            1343   
mean        2024.226000         6.499000             NaN   
min         2023.000000         1.000000             NaN   
25%         2024.000000         3.000000             NaN   
50%         2024.000000         7.000000             NaN   
75%         2025.000000        10.000000             NaN   
max         2025.000000        12.000000             NaN   
std            0.667942         3.499875             NaN   

        Cantidad_Herramientas  
count             2000.000000  
unique                    NaN  


### 3.   Carga



In [71]:
# Ruta de destino en mi Google Drive
ruta_salida = '/content/drive/MyDrive/Analisis de datos 2025/Tendencias del mercado laboral de IA/ai_job_market_limpio.csv'

# Guardado de DataFrame limpio
df.to_csv(ruta_salida, index=False)

print(" Dataset limpio guardado correctamente.")


 Dataset limpio guardado correctamente.


## 2 - EDA Análisis Exploratorio de datos

In [72]:
df['Puesto'].value_counts().head(10)
df['Industria'].value_counts()
df['Experiencia'].value_counts()

#Entry (nivel inicial) perfiles junior
#Mid y Senior: diversificado en niveles de experiencia.

#¿Qué nivel de experiencia es más demandado?
#Conclusión: Las empresas buscan más perfiles Entry-level,
#lo que indica alta demanda de talento junior en el mercado de IA.

Experiencia
Entry     702
Mid       668
Senior    630
Name: count, dtype: int64

In [65]:
df.groupby('Experiencia')['Salario_Promedio'].mean().round(2)

#¿Cómo varía el salario según el nivel?

# Entry-level tiene un salario promedio más alto que Mid, lo cual es inusual.
# Senior sí tiene el salario más alto, como se espera.

# La diferencia entre niveles es muy pequeña, lo que sugiere que:
# Puede haber ofertas junior con salarios inflados (por ubicación, industria o herramientas).
# O que Mid-level está subvalorado en algunas empresas.


Experiencia
Entry     123404.26
Mid       121441.44
Senior    124329.10
Name: Salario_Promedio, dtype: float64

In [69]:
from collections import Counter

# ¿Qué herramientas son más solicitadas por nivel?

# Función para contar herramientas por nivel
def herramientas_por_nivel(nivel):
    herramientas = df[df['Experiencia'] == nivel]['Herramientas'].dropna().str.split(',')
    flat_list = [h.strip() for sublist in herramientas for h in sublist if isinstance(sublist, list)]
    return Counter(flat_list).most_common(5)

niveles = ['Entry', 'Mid', 'Senior']
for nivel in niveles:
    print(f"\n -> Herramientas más pedidas en nivel {nivel}:")
    for herramienta, cantidad in herramientas_por_nivel(nivel):
        print(f"- {herramienta}: {cantidad} veces")

# Conclusión general
# Herramientas como LangChain, FastAPI y KDB+ son transversales a todos los niveles.
# A medida que aumenta la experiencia, se incorporan herramientas más complejas como MLflow, TensorFlow y Scikit-learn.
# Esto refleja una progresión técnica clara: desde el acceso y despliegue hasta el modelado y gestión avanzada.


 ->Herramientas más pedidas en nivel Entry:
- BigQuery: 192 veces
- LangChain: 176 veces
- FastAPI: 176 veces
- PyTorch: 175 veces
- KDB+: 174 veces

 ->Herramientas más pedidas en nivel Mid:
- LangChain: 181 veces
- MLflow: 178 veces
- FastAPI: 172 veces
- KDB+: 171 veces
- TensorFlow: 161 veces

 ->Herramientas más pedidas en nivel Senior:
- Scikit-learn: 173 veces
- MLflow: 170 veces
- TensorFlow: 162 veces
- FastAPI: 157 veces
- KDB+: 154 veces


In [73]:
# ¿Qué industrias lideran la contratación en IA?
# Automoviles
df['Industria'].value_counts().head()

Industria
Automotive    300
Education     294
Retail        293
E-commerce    291
Finance       279
Name: count, dtype: int64

# Documentación

### 1. ETL – Extracción, Transformación y Carga y  EDA

El objetivo de esta sección es conocer, visualizar y comprender las relaciones dentro del dataset de ofertas laborales en inteligencia artificial.

---

###  Extracción

- Se importó el dataset desde un archivo `.csv` ubicado en Google Drive:
  ```python
  ruta_archivo = '/content/drive/MyDrive/Analisis de datos 2025/Tendencias del mercado laboral de IA/ai_job_market.csv'
  df = pd.read_csv(ruta_archivo)

###  Tranformacion
- Se explica mas adelante
###  Carga
- El DataFrame limpio se guardó como archivo .csv para su análisis posterior:

  ```
    df.to_csv('/content/drive/MyDrive/Analisis de datos 2025/Tendencias del mercado laboral de IA/ai_job_market_limpio.csv', index=False)
  ```
    
---
---

###  Estructura general del dataset

- **Cantidad de filas:** 2000 (ejemplo)
- **Cantidad de columnas:** 12
- **Tipos de variables:**
  - Numéricas: `Salario_Min`, `Salario_Max`, `Salario_Promedio`, `Cantidad_Herramientas`
  - Categóricas: `Puesto`, `Industria`, `Experiencia`, `Tipo_Empleo`, `Ubicación`
  - Texto libre: `Herramientas`
  - Fecha: `Fecha_Publicación`

---

###  Análisis con `df.info()` y `df.describe(include='all')`

- Se identificaron columnas con valores faltantes, que fueron tratadas en la etapa de transformación.
- Las columnas categóricas como `Experiencia` y `Tipo_Empleo` mostraron valores frecuentes útiles para segmentar el análisis.
- Las columnas numéricas (`Salario_Promedio`, `Cantidad_Herramientas`) presentaron valores consistentes, sin outliers extremos.

---

---
### 3. Preguntas de Negocio

#### 🔹¿Qué nivel de experiencia es más demandado?

| Nivel de experiencia | Cantidad de ofertas |
|----------------------|---------------------|
| Entry                | 702                 |
| Mid                  | 668                 |
| Senior               | 630                 |

> Las empresas buscan principalmente perfiles **Entry-level**, lo que indica alta demanda de talento junior.

---

#### 🔹 ¿Cómo varía el salario según el nivel de experiencia?

| Nivel     | Salario Promedio (USD) |
|-----------|------------------------|
| Entry     | 123,404.26             |
| Mid       | 121,441.44             |
| Senior    | 124,329.10             |

**Respuesta:**  
Aunque se espera que el salario aumente con la experiencia, el nivel **Entry** presenta un salario promedio elevado. Esto puede deberse a:
- Ofertas junior con salarios inflados por ubicación o herramientas específicas
- Subvaloración de perfiles Mid en algunas industrias

---

#### 🔹 ¿Qué herramientas son más solicitadas por nivel?

**Entry:**
- BigQuery, LangChain, FastAPI, PyTorch, KDB+

**Mid:**
- LangChain, MLflow, FastAPI, KDB+, TensorFlow

**Senior:**
- Scikit-learn, MLflow, TensorFlow, FastAPI, KDB+

**Respuesta:**  
Herramientas como **FastAPI** y **KDB+** son transversales a todos los niveles. A medida que aumenta la experiencia, se incorporan herramientas más complejas como **MLflow**, **TensorFlow** y **Scikit-learn**.

---

#### 🔹 ¿Qué industrias lideran la contratación en IA?

| Industria     | Cantidad de ofertas |
|---------------|---------------------|
| Automotive    | 300                 |
| Education     | 294                 |
| Retail        | 293                 |
| E-commerce    | 291                 |
| Finance       | 279                 |

**Respuesta:**  
La industria **automotriz** lidera la contratación en IA, seguida muy de cerca por educación, retail y e-commerce. Esto refleja una adopción transversal de la inteligencia artificial en sectores tanto tradicionales como digitales.

---
---


## Conclusión

El análisis exploratorio permitió identificar patrones clave en el mercado laboral de IA:
- Alta demanda de perfiles junior
- Progresión salarial poco lineal
- Herramientas específicas asociadas a cada nivel de experiencia
- La industria **automotriz** lidera la contratación en IA

Estos hallazgos serán fundamentales para responder las preguntas de negocio.
